src/pico/i_video.c

```diff
@@ -949,108 +797,102 @@ void __noinline new_frame_init_overlays_palette_and_wipe() {
 
+    for (int scanline = 0; scanline < end_scanline; scanline++){
         DEBUG_PINS_SET(scanline_copy, 1);
-        if (display_video_type != VIDEO_TYPE_TEXT) {
-            // we don't have text mode -> normal transition yet, but we may for network game, so leaving this here - we would need to put the buffer pointers back
-            assert (buffer->data < text_scanline_buffer_start || buffer->data >= text_scanline_buffer_start + TEXT_SCANLINE_BUFFER_TOTAL_WORDS);
-            scanline_funcs[display_video_type](buffer->data+1, scanline);
-            if (display_video_type >= FIRST_VIDEO_TYPE_WITH_OVERLAYS) {
-                assert(scanline < count_of(vpatchlists->vpatch_starters));
-                int prev = 0;
-                for (int vp = vpatchlists->vpatch_starters[scanline]; vp;) {
-                    int next = vpatchlists->vpatch_next[vp];
-                    while (vpatchlists->vpatch_next[prev] && vpatchlists->vpatch_next[prev] < vp) {
-                        prev = vpatchlists->vpatch_next[prev];
-                    }
-                    assert(prev != vp);
-                    assert(vpatchlists->vpatch_next[prev] != vp);
-                    vpatchlists->vpatch_next[vp] = vpatchlists->vpatch_next[prev];
-                    vpatchlists->vpatch_next[prev] = vp;
-                    prev = vp;
-                    vp = next;
-                }
-                vpatchlist_t *overlays = vpatchlists->overlays[display_overlay_index];
-                prev = 0;
-                for (int vp = vpatchlists->vpatch_next[prev]; vp; vp = vpatchlists->vpatch_next[prev]) {
-                    patch_t *patch = resolve_vpatch_handle(overlays[vp].entry.patch_handle);
-                    int yoff = scanline - overlays[vp].entry.y;
-                    if (yoff < vpatch_height(patch)) {
-                        vpatchlists->vpatch_doff[vp] = draw_vpatch((uint16_t*)(buffer->data + 1), patch, &overlays[vp],
-                                                                   vpatchlists->vpatch_doff[vp]);
-                        prev = vp;
-                    } else {
-                        vpatchlists->vpatch_next[prev] = vpatchlists->vpatch_next[vp];
-                    }
-                }
-            }
-            uint16_t *p = (uint16_t *) buffer->data;
-            p[0] = video_doom_offset_raw_run;
-            p[1] = p[2];
-            p[2] = SCREENWIDTH - 3;
-            buffer->data[SCREENWIDTH / 2 + 1] = video_doom_offset_raw_1p;
-            buffer->data[SCREENWIDTH / 2 + 2] = video_doom_offset_end_of_scanline_skip_ALIGN;
-            buffer->data_used = SCREENWIDTH / 2 + 3;
-            DEBUG_PINS_CLR(scanline_copy, 1);
-        } else {
+        switch(display_video_type) {
+            case VIDEO_TYPE_NONE :
+                scanline_func_none(buffer, scanline);
+                break;
+            case VIDEO_TYPE_SINGLE : 
+                scanline_func_single(buffer, scanline);
```

Comment:
I started to have trouble with function pointers - not sure what was going on, but switching to a swtich fixed it, so I never went back